addons/pos_self_order/models/pos_config.py

```diff
@@ -29,6 +83,72 @@ def preview_self_order_app(self):
         self.ensure_one()
         return {
             "type": "ir.actions.act_url",
-            "url": f"/menu/{self.id}",
+            "url": self._get_self_order_route(),
             "target": "new",
         }
+
+    def _get_self_order_custom_links(self) -> List[Dict[str, str]]:
+        """
+        On the landing page of the app we can have a number of custom links
+        that are defined by the restaurant employee in the backend.
+        This function returns a list of dictionaries with the attributes of each link
+        that is available for the POS with id pos_config_id.
+        """
+        self.ensure_one()
+        return (
+            self.env["pos_self_order.custom_link"]
+            .sudo()
+            .search_read(
+                [
+                    "|",
+                    ("pos_config_ids", "in", [self.id]),
+                    ("pos_config_ids", "=", False),
+                ],
+                fields=["name", "url", "style"],
+                order="sequence",
+            )
+        )
+
+    def _get_available_products(self) -> ProductProduct:
```

Comment:
I just noticed that this method is just as worse as the loading of products in pos. Can't we lazy-load the products? Something like fetch-on-scroll or some pagination?

The thing is, if you do this, then why don't we just use the original pos app?

Reply:
Well, I guess we can assume that pos users will setup their pos to only load limited number of products.

Maybe later we will provide some overview of how many products will be loaded when the customers open the self-order web app.